src/lib/elements/Input/Input.svelte

```diff
@@ -45,28 +58,73 @@
         return true
     }
 
-    if (copyOnInteract) {
-        tooltip = "Copy"
-        disabled = true
+    // Debounced validation function
+    const debouncedValidateInput = debounce(() => {
+        let isValid = isValidInput()
+        dispatch("isValid", isValid)
+    }, 300);
+
+    // Handle input event
+    function handleInput(event) {
+        value = event.target.value
+        debouncedValidateInput()
     }
 
-    let clazz = ""
-    let input: HTMLElement
-    const dispatch = createEventDispatcher()
-    const writableValue = writable(value)
+    // Handle paste event
+    function handlePaste(event) {
+        setTimeout(() => {
+            value = event.target.value
+            debouncedValidateInput()
+        }, 0)  // Delay to ensure the pasted value is updated
+    }
+
+    // Handle input change
+    function onInput() {
+        let isValid = isValidInput()
+        dispatch("isValid", isValid)
+        dispatch("input", value)
+    }
 
-    $: writableValue.set(value)
-    $: value = $writableValue
+    function onBlur() {
+        dispatch("blur");
```

Comment:
we should remove the semicolons `;` from the line endings to match the style guide

Reply:
fosho! Removed :tada: was missing that one :joy: